docs/check_spelling.sh

```diff
@@ -0,0 +1,185 @@
+#!/bin/bash
+set -e
+
+# This script does spell checks and can add words to a dictionary.
+# It can be used as a git hook to perform checks, it should be named 
+# "pre-push","pre-commit" based on the use case and kept into the ".git/hooks/"
+# and be executable.It can be bypassed by--no-verify option with git command.
+
+# Language of the doc, install appropriate aspell libs when using 
+# a non-English language
+lang=en
+
+project_dict="./.aspell.en.pws"
+
+# Extension for files to be spell checked
+file_extension=rst
+
+# Default execution mode for the script, check usage for more options
+mode="one"
+
+# Use aspell with no backup mode
+aspell_cmd="aspell --dont-backup --personal=$project_dict --lang=$lang"
+
+custom_dicts=""
+
+misspelled_wordlist=()
```

Comment:
Is this for autocorrect? A comment wouldn't hurt here either.

Reply:
Yes it is. Updated comments.